homeassistant/components/webostv/__init__.py

```diff
@@ -98,55 +157,61 @@ async def async_service_handler(service):
             DOMAIN, service, async_service_handler, schema=schema
         )
 
-    tasks = [async_setup_tv(hass, config, conf) for conf in config[DOMAIN]]
-    if tasks:
-        await asyncio.gather(*tasks)
+    hass.data[DOMAIN][host] = {"client": client}
 
-    return True
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, component)
+        )
 
+    # set up notify platform, no entry support for notify component yet,
+    # have to use discovery to load platform.
+    hass.async_create_task(
+        discovery.async_load_platform(
+            hass,
+            "notify",
+            DOMAIN,
+            {
+                CONF_HOST: host,
+                CONF_ICON: config_entry.data.get(CONF_ICON),
+                CONF_NAME: config_entry.data[CONF_NAME],
+            },
+            hass.data[DOMAIN],
+        )
+    )
 
-def convert_client_keys(config_file):
-    """In case the config file contains JSON, convert it to a Sqlite config file."""
-    # Return early if config file is non-existing
-    if not os.path.isfile(config_file):
-        return
+    if not config_entry.update_listeners:
+        config_entry.add_update_listener(async_update_options)
```

Comment:
Wrap the callable this returns in entry.async_on_unload instead to make sure it doesn't leak memory on unload